frontend/src/components/Collections/components/Grid/components/CollectionsGrid/index.tsx

```diff
@@ -42,29 +56,61 @@ const CollectionsGrid: FC<Props> = ({
 function renderCollections(
   collections: CollectionResponse[],
   displayVisibility?: VISIBILITY_TYPE,
-  accessType?: ACCESS_TYPE
+  accessType?: ACCESS_TYPE,
+  revisionsEnabled = false
 ) {
   const collectionElements = [] as Array<ReactChild>;
-  sortByCreatedAtDescending(collections).forEach(({ id, visibility }) => {
-    if (!displayVisibility || visibility === displayVisibility) {
-      collectionElements.push(
-        <CollectionRow
-          id={id}
-          key={id + visibility}
-          visibility={visibility}
-          accessType={accessType}
-        />
-      );
+  const revisions = generateRevisions(collections, revisionsEnabled);
+  sortByCreatedAtDescending(revisions).forEach(
+    ({ id, visibility, revision }) => {
+      if (!displayVisibility || visibility === displayVisibility) {
+        collectionElements.push(
+          <CollectionRow
+            id={id}
+            key={id + visibility}
+            visibility={visibility}
+            accessType={accessType}
+            revisionStatus={revision}
+          />
+        );
+      }
     }
-  });
+  );
 
   return collectionElements;
 }
 
 function sortByCreatedAtDescending(
-  collections: CollectionResponse[]
-): CollectionResponse[] {
+  collections: RevisionResponse[]
+): RevisionResponse[] {
   return collections?.sort((a, b) => b.created_at - a.created_at) || [];
 }
 
+function generateRevisions(
+  collections: CollectionResponse[],
+  revisionsEnabled: boolean
+): RevisionResponse[] {
+  const newCollections = collections.map(
+    (collection): RevisionResponse => {
+      return { ...collection, revision: REVISION_STATUS.DISABLED };
+    }
+  );
+  // If revisions are disabled just return the array with objects' revisions status set to disabled
+  if (!revisionsEnabled) return newCollections;
+
+  const revisionMap = new Map<string, RevisionResponse>();
+
+  newCollections.forEach((collection) => {
+    const revisionObj = revisionMap.get(collection.id) || collection;
+    if (revisionObj.revision !== REVISION_STATUS.STARTED) {
+      revisionObj.revision =
+        revisionObj.revision === REVISION_STATUS.DISABLED
```

Comment:
ahh I see, so if a public collection the user owns has a revision, it will have the same id with `visibility === PRIVATE` , so the second time the loop encounters the same id, it will bump the revision enum from `NOT_STARTED` to `STARTED`? 

That works! But I wonder if there's a more explicit way of doing it. Not a blocker though!

Reply:
That's it! We could probably ask for a revision property in the future. I think a /user/collections endpoint would be higher priority though